pymsis/utils.py

```diff
@@ -207,7 +280,7 @@ def get_f107_ap(dates: npt.ArrayLike) -> tuple[npt.NDArray, npt.NDArray, npt.NDA
             |     prior to current time
     """
     dates = np.asarray(dates, dtype=np.datetime64)
-    data = _DATA or _load_f107_ap_data()
+    data = _DATA or _load_f107_ap_data(dates[-1])
```

Comment:
If we have a long running process with `_DATA` already loaded, then this won't update the data for us. I think the logic should probably be a new if block here instead after the original loading potentially?
```suggestion
    # Load/download data right away.
    data = _DATA or _load_f107_ap_data()
    if dates[-1] > data["dates"][-1]:
        data = _new_function_to_download_recent_data()
```